pre_commit_hooks/check_case_conflict.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+
+from plumbum import local
+
+from pre_commit_hooks.util import added_files
+
+
+def find_conflicting_filenames(filenames):
+    repo_files = set(local['git']('ls-files').splitlines())
+    relevant_files = set(filenames) | added_files()
+    repo_files -= relevant_files
+    retv = 0
+
+    # new file conflicts with existing file
+    conflicts = (set(x.lower() for x in repo_files)
+                 & set(x.lower() for x in relevant_files))
```

Comment:
just a nit:

I usually prefer this written as:

conficts = (
    (...) &
    (...)
)